core/services/workflows/syncer/v2/workflow_registry.go

```diff
@@ -320,7 +320,10 @@ func (w *workflowRegistry) generateReconciliationEvents(_ context.Context, pendi
 	}
 
 	if len(pendingEvents) != 0 {
-		return nil, fmt.Errorf("invariant violation: some pending events were not handled in the reconcile loop: keys=%+v, len=%d", maps.Keys(pendingEvents), len(pendingEvents))
+		w.lggr.Debugf("some pending events were not handled in the reconcile loop, cleaning up: keys=%+v, len=%d", maps.Keys(pendingEvents), len(pendingEvents))
```

Comment:
@justinkaseman Hmmm... the intention was for the code above to handle all cases, including cleaning up any pending events that are no longer relevant. (See for instance this clause which illustrates what I mean: https://github.com/smartcontractkit/chainlink/pull/19117/files#diff-66829056a38220ab64391f36184d372d6ec0e3c03ef7ae393f103ecd378b1d92L268)

Do you have an example of a scenario that wasn't handled by the code above?